js/logo.js

```diff
@@ -4345,6 +4345,22 @@ function Logo () {
                 that._setListener(turtle, listenerName, __listener);
             }
             break;
+		case 'noise':
+			if (args[0] == null) {
+                that.errorMsg(NOINPUTERRORMSG, blk);
+                childFlow = args[1];
+                childFlowCount = 1;
+			} else if (args[0] == "white" || args[0] == "pink" || args[0] == "brown") {
+				var noiseSynth = new Tone.NoiseSynth().toMaster();
+				noiseSynth.set("noise.type", args[0]);
+				noiseSynth.triggerAttackRelease("2");
```

Comment:
Why "2"? What we really want it to be able to use the noise synth inside of a note value block so the duration is determined by the note value. So you'll probably want to move this part of the code to musicutils.js and treat the noise synth like the other synths.

Reply:
I tried but the noise synthesizer does not seem to be able to integrate into musicutils.js unlike the AM and FM synthesizers; however the note value can be included as a separate argument to the noise block.